shell/browser/api/atom_api_web_contents.cc

```diff
@@ -346,6 +350,23 @@ base::Optional<base::TimeDelta> GetCursorBlinkInterval() {
   return base::nullopt;
 }
 
+#if BUILDFLAG(ENABLE_PRINTING)
+// This will return false if no printer with the provided device_name can be
+// found on the network. We need to check this because Chromium does not do
+// sanity checking of device_name validity and so will crash on invalid names.
+bool DeviceNameValid(const std::string& device_name) {
+#if defined(OS_MACOSX)
+  base::ScopedCFTypeRef<CFStringRef> new_printer_id(
+      base::SysUTF8ToCFStringRef(device_name));
```

Comment:
`SysUTF16ToCFStringRef` can be used here so `DeviceNameValid` can take `base::string16` as parameter to avoid unnecessary conversions on Windows.